packages/nextjs/components/Mapbox.js

```diff
@@ -1,30 +1,62 @@
-import mapboxgl from "!mapbox-gl";
 // eslint-disable-line import/no-webpack-loader-syntax
-import React, { useRef, useEffect, useState } from "react";
+import React, { useEffect, useRef, useState } from "react";
+import mapboxgl from "!mapbox-gl";
+import "mapbox-gl/dist/mapbox-gl.css";
 
 mapboxgl.accessToken = "pk.eyJ1Ijoicm9uY2h1Y2siLCJhIjoiY2x2Y2o5Z2drMGY3cjJrcGI4b2xsNzdtaCJ9.gi5RJ8qRhTSwfYuhVwhmvQ";
-export default function Mapbox() {
+export default function Mapbox({ setCheckInActive }) {
   const mapContainer = useRef(null);
   const map = useRef(null);
-  const [lng, setLng] = useState(-118.464182);
-  const [lat, setLat] = useState(33.982547);
-  const [zoom, setZoom] = useState(14);
+  const markersRef = useRef([]);
+  const [lng, setLng] = useState(-104.98716);
+  const [lat, setLat] = useState(39.7323862);
+  const [zoom, setZoom] = useState(15);
 
   useEffect(() => {
-    if (map.current) return;
+    if (!map.current) {
+      map.current = new mapboxgl.Map({
+        container: mapContainer.current,
+        style: "mapbox://styles/mapbox/dark-v11",
+        center: [lng, lat],
+        zoom: zoom,
+      });
+    }
     // initialize map only once
-    map.current = new mapboxgl.Map({
-      container: mapContainer.current,
-      style: "mapbox://styles/mapbox/dark-v11",
-      center: [lng, lat],
-      zoom: zoom,
+
+    map.current?.on("move", () => {
+      setLng(map.current.getCenter().lng.toFixed(4));
+      setLat(map.current.getCenter().lat.toFixed(4));
+      setZoom(map.current.getZoom().toFixed(2));
+    });
+    map.current?.on("click", e => {
+      // Clear existing markers
+      markersRef.current.forEach(marker => marker.remove());
+      markersRef.current = [];
+
+      // Set state variables
+      setLng(e.lngLat.lng);
+      setLat(e.lngLat.lat);
+
+      // Add a pin to the map
+      var newMarker = new mapboxgl.Marker({
+        color: "red",
+        // anchor: "bottom",
```

Comment:
add a button (check-in, and cancel) like so?
https://docs.mapbox.com/help/tutorials/custom-markers-gl-js/#attach-popups-to-markers

Reply:
Up to you, but I think it's easier to only create a button component at the button, and in a separate pr attach it to the marker.